File: app/src/main/java/com/suzei/racoon/ui/profile/ProfilePresenter.java
package com.suzei.racoon.ui.profile;

import com.suzei.racoon.model.Users;
import com.suzei.racoon.ui.base.Contract;
import com.suzei.racoon.ui.profile.ProfileInteractor;

public class ProfilePresenter implements Contract.Listener<Users> {

    private Contract.DetailsView<Users> profileView;
    private ProfileInteractor profileInteractor;

    public ProfilePresenter(Contract.DetailsView<Users> profileView) {
        this.profileView = profileView;
        profileInteractor = new ProfileInteractor(this);
    }

    public void showUserDetails(String uid) {
        profileView.showProgress();
        profileInteractor.loadDetails(uid);
    }

    public void destroy(String uid) {
        profileInteractor.destroy(uid);
    }

    @Override
    public void onLoadSuccess(Users data) {
        profileView.hideProgress();
        profileView.onLoadSuccess(data);
    }

    @Override
    public void onLoadFailed(String message) {
        profileView.hideProgress();
        profileView.onLoadFailed(message);
    }
}

File: app/src/main/java/com/suzei/racoon/ui/profile/ProfilePresenter.kt
package com.suzei.racoon.ui.profile

import com.suzei.racoon.model.Users
import com.suzei.racoon.ui.base.Contract
import com.suzei.racoon.ui.base.Contract.DetailsView

class ProfilePresenter(
    private val profileView: DetailsView<Users>
) : Contract.Listener<Users> {
    private val profileInteractor: ProfileInteractor = ProfileInteractor(this)

    fun showUserDetails(uid: String) {
        profileView.showProgress()
        profileInteractor.loadDetails(uid)
    }

    fun destroy(uid: String) {
        profileInteractor.destroy(uid)
    }

    override fun onLoadSuccess(data: Users) {
        profileView.hideProgress()
        profileView.onLoadSuccess(data)
    }

    override fun onLoadFailed(message: String) {
        profileView.hideProgress()
        profileView.onLoadFailed(message)
    }

}